model/registry/gen-ai.yaml

```diff
@@ -0,0 +1,72 @@
+groups:
+  - id: registry.llm
+    prefix: gen_ai
+    type: attribute_group
+    brief: >
+      This document defines the attributes used to describe telemetry in the context of LLM (Large Language Models) requests and responses.
+    attributes:
+      - id: system
+        type:
+          allow_custom_values: true
+          members:
+            - id: openai
+              value: "OpenAI"
+              brief: 'OpenAI models like GPT, DALL-E, Sora, etc.'
+        brief: The name of the LLM foundation model vendor, if applicable.
+        examples: 'openai'
+        tag: llm-generic-request
+      - id: request.model
+        type: string
+        brief: The name of the LLM a request is being made to.
+        examples: 'gpt-4'
+        tag: llm-generic-request
+      - id: request.max_tokens
+        type: int
+        brief: The maximum number of tokens the LLM generates for a request.
+        examples: [100]
+        tag: llm-generic-request
+      - id: request.temperature
+        type: double
+        brief: The temperature setting for the LLM request.
+        examples: [0.0]
+        tag: llm-generic-request
+      - id: request.top_p
+        type: double
+        brief: The top_p sampling setting for the LLM request.
+        examples: [1.0]
+        tag: llm-generic-request
+      - id: response.id
+        type: string
+        brief: The unique identifier for the completion.
+        examples: ['chatcmpl-123']
+        tag: llm-generic-response
+      - id: response.model
+        type: string
+        brief: The name of the LLM a response is being made to.
+        examples: ['gpt-4-0613']
+        tag: llm-generic-response
+      - id: response.finish_reasons
+        type: string[]
+        brief: Array of reasons the model stopped generating tokens, corresponding to each generation received.
+        examples: [['stop']]
+        tag: llm-generic-response
+      - id: usage.prompt_tokens
+        type: int
+        brief: The number of tokens used in the LLM prompt.
+        examples: [100]
+        tag: llm-generic-response
+      - id: usage.completion_tokens
+        type: int
+        brief: The number of tokens used in the LLM response (completion).
+        examples: [180]
+        tag: llm-generic-response
+      - id: prompt
+        type: string
+        brief: The full prompt sent to an LLM, as a stringified JSON in OpenAI's format.
```

Comment:
Given the ambiguity we have on how to describe events and their structure (#834, #829), maybe we can be less specific here in this PR and add details later?
E.g. in the following way:
```suggestion
        brief: The full prompt sent to an LLM.
```

We can also add a note with something like: "It's RECOMMENDED to format prompts as JSON string matching [OpenAI messages format](https://platform.openai.com/docs/guides/text-generation)"

The subtle difference is that we're not guaranteeing it for the telemetry consumers. And instrumentations can do something else if they have good reasons to.

JFYI: I believe this bug https://github.com/open-telemetry/build-tools/issues/305 prevents you from adding notes on the original attributes definitions, you can keep adding notes on the referenced attributes, or you can generate tables with `<!-- semconv <group id>(full) -->`